### PR TITLE
Another round of Dialyzer work.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Removes almost all but the remaining 15 Dialyzer errors.  Each fix is it's own commit and contains the Dialyzer errors removed.

Tagging @seancribbs and @andrewjstone for reviews.

### CODE REVIEW

File: src/riak_core_cluster_mgr.erl [hunk 1]
```diff
@@ -710,7 +710,7 @@ persist_members_to_ring(State, ClusterName, Members) ->
 %% that were saved in the ring
 cluster_mgr_sites_fun() ->
     %% get cluster names from cluster manager
-    Ring = riak_core_ring_manager:get_my_ring(),
+    {ok, Ring} = riak_core_ring_manager:get_my_ring(),
```
Comment (bsparrow435) [inline src/riak_core_cluster_mgr.erl:713]: wow, good catch. Amazing this hasn't caused issues in the past

Comment (cmeiklejohn) [inline src/riak_core_cluster_mgr.erl:713]: We've flattened a bunch of the locator stuff, so it's probably used in very few cases at this point.

File: src/riak_repl_fullsync_worker.erl [hunk 2]
```diff
@@ -78,7 +79,7 @@ handle_call({get, B, K, Transport, Socket, Pool, Partition, Ver}, From, State) -
 
     Preflist = [{Partition, OwnerNode}],
 
-    ReqID = make_ref(),
+    ReqID = make_req_id(),
```
Comment (bsparrow435) [inline src/riak_repl_fullsync_worker.erl:82]: Wow, another great catch.

**Comment by seancribbs** [comment]
Other than the one comment I left on 60987bc, I'm ok with this. Good work. :+1:

**Comment by andrewjstone** [comment]
replication2 passes
 :+1:

**Pull request merged**